src/create/useFormScoresCreate.js

```diff
@@ -0,0 +1,179 @@
+import { useEffect, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import {
+    validateScoreIsInRange
+} from '../app/validators.services'
+import loadLocally from '../lib/loadLocally'
+import removeLocally from '../lib/removeLocally'
+import saveLocally from '../lib/saveLocally'
+
+const STORAGE_KEY = 'inputsScoresCreate'
+
+export default function useFormScoresCreate({
+    newGameProfile,
+    addGameProfile,
+    updateVisibleForm }) {
+    
+    const history = useHistory()
+    
+    const [inputsScores, setInputsScores] = useState(loadLocally(STORAGE_KEY) ?? newGameProfile)
+
+    const validInputs = inputsScores.playersArray && prepareInputValidation(inputsScores)
+    const hasAllValidInputs = validInputs && condenseValidInputs()
+    
+    const isSaveButtonShownScores = hasAllValidInputs
+
+    const [dirtyInputs, setDirtyInputs] = useState({})
+    const hasAllDirtyInputs = dirtyInputs && condenseDirtyInputs()
+
+    useEffect(() => saveLocally(STORAGE_KEY, inputsScores), [inputsScores])
+    useEffect(() => setInputsScores(newGameProfile), [newGameProfile])
+    useEffect(() => prepareInputValidation(newGameProfile), [newGameProfile])
+    useEffect(() => prepareDirtyInputs(newGameProfile), [newGameProfile])
```

Comment:
Can't you combine these three useEffects for `newGameProfile`?

```
    useEffect(() => { 
    setInputsScores(newGameProfile)
    prepareInputValidation(newGameProfile)
    prepareDirtyInputs(newGameProfile)
    } , [newGameProfile])
```

Reply:
I'll check this again. I have tried it before, but it would not work properly.